packages/core-data/src/types/comment.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * Internal dependencies
+ */
+import {
+	AvatarUrls,
+	Context,
+	ContextualField,
+	RawField,
+	EditedRecord,
+} from './common';
+
+interface FullComment< C extends Context > {
+	/**
+	 * Unique identifier for the comment.
+	 */
+	id: number;
+	/**
+	 * The ID of the user object, if author was a user.
+	 */
+	author: number;
+	/**
+	 * Email address for the comment author.
+	 */
+	author_email: ContextualField< string, 'edit', C >;
+	/**
+	 * IP address for the comment author.
+	 */
+	author_ip: ContextualField< string, 'edit', C >;
```

Comment:
I like the way the definitions here mimic the definitions inside the schema and documentation for the endpoints.